### PR TITLE
[erigon2] reduced memory footprint on building huffman table

### PR DESCRIPTION
**Pull request by awskii**
another optimisation to reduce RAM required for huffman tables up to 45% in compare with a86660187debb8362458892e3ad877c2ec002775 .

### CODE REVIEW

File: compress/decompress.go [hunk 1]
```diff
@@ -372,9 +378,9 @@ func (g *Getter) HasNext() bool {
 // After extracting next word, it moves to the beginning of the next one
 func (g *Getter) Next(buf []byte) ([]byte, uint64) {
 	savePos := g.dataP
-	l := g.nextPos(true)
-	l-- // because when create huffman tree we do ++ , because 0 is terminator
-	if l == 0 {
+	nextPos := g.nextPos(true)
```
Comment (AskAlexSharov) [inline compress/decompress.go:381]: are you sure this is nextPos and not lenght of word?

**Comment by AskAlexSharov** [comment]
BSC `go tool pprof -inuse_space -png http://127.0.0.1:6060/debug/pprof/heap > mem.png`
mem

File: compress/decompress.go (hunk 1 above)
Comment (awskii) [reply compress/decompress.go:381]: not sure, but nextPos looks cleaner than some `l` defined at the function's beginning. could rename to `wordLength` or `patternLength`

Comment (AskAlexSharov) [reply compress/decompress.go:381]: yes, please rename - to something that shows real meaning of variable (I also don't know - what it means :-) )

**Pull request merged**